community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/Dijkstra.java

```diff
@@ -51,74 +58,159 @@
+                                         PathInterestFactory.all( Precision.EPSILON );
+        epsilon = Precision.EPSILON;
+        this.stateInUse = true;
     }
 
-    public Dijkstra( RelationshipExpander expander, CostEvaluator<Double> costEvaluator )
+    public Dijkstra( PathExpander expander, CostEvaluator<Double> costEvaluator )
     {
-        this( toPathExpander( expander ), costEvaluator, true );
+        this( expander, costEvaluator, PathInterestFactory.allShortest( Precision.EPSILON ) );
     }
 
-    public Dijkstra( PathExpander expander, CostEvaluator<Double> costEvaluator, boolean stopAfterLowestCost )
+    /**
+     * @deprecated in favor for {@link #Dijkstra(PathExpander, CostEvaluator)}
+     * @param expander
+     * @param costEvaluator
+     */
+    public Dijkstra( RelationshipExpander expander, CostEvaluator<Double> costEvaluator )
     {
-        this( expander, InitialBranchState.NO_STATE, costEvaluator, stopAfterLowestCost );
+        this( toPathExpander( expander ), costEvaluator );
     }
 
-    public Dijkstra( PathExpander expander, InitialBranchState stateFactory, CostEvaluator<Double> costEvaluator,
+    /**
+     * @deprecated in favor for {@link #Dijkstra(PathExpander, CostEvaluator, PathInterest)}  }.
+     * @param expander
+     * @param costEvaluator
+     * @param stopAfterLowestCost
+     */
+    public Dijkstra( PathExpander expander, CostEvaluator<Double> costEvaluator,
             boolean stopAfterLowestCost )
     {
-        this.expander = expander;
-        this.costEvaluator = costEvaluator;
-        this.stateFactory = stateFactory;
-        this.stopAfterLowestCost = stopAfterLowestCost;
+        this( expander, costEvaluator, Precision.EPSILON, stopAfterLowestCost ?
+                                                          PathInterestFactory.allShortest( Precision.EPSILON ) :
+                                                          PathInterestFactory.all( Precision.EPSILON ) );
     }
 
+
+    /**
+     * @deprecated in favor for {@link #Dijkstra( PathExpander, CostEvaluator, PathInterest ) }.
+     * @param expander
+     * @param costEvaluator
+     * @param stopAfterLowestCost
+     */
     public Dijkstra( RelationshipExpander expander, CostEvaluator<Double> costEvaluator, boolean stopAfterLowestCost )
     {
         this( toPathExpander( expander ), costEvaluator, stopAfterLowestCost );
     }
 
+    public Dijkstra( PathExpander expander, CostEvaluator<Double> costEvaluator, PathInterest<Double> interest )
+    {
+        this( expander, costEvaluator, Precision.EPSILON, interest );
+    }
+
+    public Dijkstra( PathExpander expander, CostEvaluator<Double> costEvaluator, double epsilon,
+            PathInterest<Double> interest )
+    {
+        this.expander = expander;
+        this.costEvaluator = costEvaluator;
+        this.epsilon = epsilon;
+        this.interest = interest;
+        this.stateFactory = new InitialBranchState<Double>()
```

Comment:
Same `InitialBranchState` used at https://github.com/neo4j/neo4j/pull/4879/files#diff-8a50a6e422c7dbd65844eed8a06cc7c1R89
Does it make sense to add something similar to `InitialBranchState.NO_STATE` fro those cases?